exporters/otlp/src/otlp_environment.cc

```diff
@@ -80,6 +80,34 @@ static bool GetStringDualEnvVar(const char *signal_name,
   return exists;
 }
 
+static std::uint32_t GetUintEnvVarOrDefault(opentelemetry::nostd::string_view signal_env,
```

Comment:
Code changes in this file are also present in #3223

Planning to review and merge the OTLP HTTP PR first.

Reply:
All changes from the above PR were backported here, as well.

The only minor annoyance with the current use of the WITH_OTLP_RETRY_PREVIEW flag on CI is that none of the maintainer builds have WITH_OTLP_GRPC enabled, so the new code would not be unit tested.

Would it be reasonable also enabling this new feature flag, for instance with "cmake.exporter.otprotocol.test", in order to have at least one build that exercises this code?

Reply:
> All changes from the above PR were backported here, as well.
> 
> The only minor annoyance with the current use of the WITH_OTLP_RETRY_PREVIEW flag on CI is that none of the maintainer builds have WITH_OTLP_GRPC enabled, so the new code would not be unit tested.
> 
> Would it be reasonable also enabling this new feature flag, for instance with "cmake.exporter.otprotocol.test", in order to have at least one build that exercises this code?

Sure, feel free to add it where it makes sense.

We should add WITH_OTLP_GRPC to the maintainer builds as well (in a separate PR), these builds are meant to have a maximum number of features covered, and should also execute the new gRPC functional tests.